static/js/disaster_dashboard/data_fetcher.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Functions that return promises for data needed for disaster dashbaord
+ */
+
+import axios from "axios";
+
+import { GeoJsonData } from "../chart/types";
+import { DISASTER_EVENT_INTENSITIES, DISASTER_EVENT_TYPES } from "./constants";
+import { DisasterApiEventPoint, DisasterEventPoint, DisasterType } from "./types";
+
+const MAX_DATES = 100;
+
+/**
+ * Get promise for geojson data
+ * @param selectedPlace the enclosing place to get geojson data for
+ * @param placeType the place type to get geojson data for
+ */
+export function fetchGeoJsonData(
+  selectedPlace: string,
+  placeType: string
+): Promise<GeoJsonData> {
+  return axios
+    .get<GeoJsonData>("/api/choropleth/geojson", {
+      params: {
+        placeDcid: selectedPlace,
+        placeType: placeType,
+      },
+    })
+    .then((resp) => resp.data as GeoJsonData);
+}
+
+/**
+ * Get promise for date list
+ * @param disasterType the disaster type to get date list for
+ */
+export function fetchDateList(disasterType: DisasterType): Promise<string[]> {
+  const promises = [];
+  const disasterTypesToFetch =
+    disasterType === DisasterType.ALL
+      ? Object.keys(DISASTER_EVENT_TYPES)
+      : [disasterType];
+  for (const disasterType of disasterTypesToFetch) {
+    DISASTER_EVENT_TYPES[disasterType].forEach((eventType) => {
+      promises.push(
+        axios
+          .get<{ minDate: string, maxDate: string }>("/api/disaster-dashboard/date-range", {
+            params: {
+              eventType,
+            },
+          })
+          .then((resp) => resp.data)
+      );
+    });
+  }
+  return Promise.all(promises).then((resp) => {
+    let minDate = "";
+    let maxDate = "";
+    resp.forEach((dateRange) => {
```

Comment:
maybe add a comment this is the date range for each disaster type

Reply:
done